modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultSemanticAnalysisPhase.java

```diff
@@ -2055,58 +2061,165 @@ public void visitRegex(ERegex userRegexNode, SemanticScope semanticScope) {
+        int javaFlags = 0;
 
         for (int i = 0; i < flags.length(); ++i) {
             char flag = flags.charAt(i);
 
             switch (flag) {
                 case 'c':
-                    constant |= Pattern.CANON_EQ;
+                    javaFlags |= Pattern.CANON_EQ;
                     break;
                 case 'i':
-                    constant |= Pattern.CASE_INSENSITIVE;
+                    javaFlags |= Pattern.CASE_INSENSITIVE;
                     break;
                 case 'l':
-                    constant |= Pattern.LITERAL;
+                    javaFlags |= Pattern.LITERAL;
                     break;
                 case 'm':
-                    constant |= Pattern.MULTILINE;
+                    javaFlags |= Pattern.MULTILINE;
                     break;
                 case 's':
-                    constant |= Pattern.DOTALL;
+                    javaFlags |= Pattern.DOTALL;
                     break;
                 case 'U':
-                    constant |= Pattern.UNICODE_CHARACTER_CLASS;
+                    javaFlags |= Pattern.UNICODE_CHARACTER_CLASS;
                     break;
                 case 'u':
-                    constant |= Pattern.UNICODE_CASE;
+                    javaFlags |= Pattern.UNICODE_CASE;
                     break;
                 case 'x':
-                    constant |= Pattern.COMMENTS;
+                    javaFlags |= Pattern.COMMENTS;
                     break;
                 default:
                     throw new IllegalArgumentException("invalid regular expression: unknown flag [" + flag + "]");
             }
         }
+        return javaFlags;
+    }
 
+    private Grok compileGrok(ERegex regex, SemanticScope semanticScope) {
+        if (false == "".equals(regex.getFlags())) {
+            throw regex.createError(
+                new IllegalArgumentException("Grok doesn't support flags but got [" + regex.getFlags() + "]")
+            );
+        }
         try {
-            Pattern.compile(pattern, constant);
-        } catch (PatternSyntaxException pse) {
-            throw new Location(location.getSourceName(), location.getOffset() + 1 + pse.getIndex()).createError(
-                    new IllegalArgumentException("invalid regular expression: " +
-                            "could not compile regex constant [" + pattern + "] with flags [" + flags + "]", pse));
+            
+            Map<String, String> patternBank = semanticScope.getScriptScope().getCompilerSettings().getGrokPatternBank();
+            MatcherWatchdog watchdog = semanticScope.getScriptScope().getGrokWatchdog();
+
+            /*
+             * Shift from the "no permissions" compilation context to one that
+             * has permission to read stuff from the classpath so that jcodings,
+             * a dependency of grok, can read what it needs from the classpath. 
+             */
+            return AccessController.doPrivileged(new PrivilegedAction<Grok>() {
```

Comment:
I want more folks to look at this from a security perspective. Do we want to be extra paranoid here and only give classpath reading permissions to the code? JCodings is just lazy loading some stuff so we could try and lazy load everything we think we could possible need but it isn't the smallest thing. OTOH, I think we'd only need the utf-8 stuff because `Grok` always uses utf-8 even though JCodings supports every code page under the sun. This was the simplest thing to get working, reviewable code. So here we are.